src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/request/TemplatePathVisitor.java

```diff
@@ -286,6 +292,37 @@ private boolean canCompleteXpath(Object extradata) {
                     || (extradata instanceof Boolean && ((Boolean) extradata).booleanValue());
         }
 
+        private String completeXpath(String source, String xpath) {
+            String sepSrc = "/";
+            String sepXpath = "/";
+            if (source.indexOf(sepSrc) == -1) sepSrc = "\\.";
+            if (xpath.indexOf(sepXpath) == -1) sepXpath = "\\.";
```

Comment:
Not sure what you're trying to do here.... the dot becomes the separator? For paths like "a.b.c?". But at the same time, you're trying to handle ".." in xpath....